### PR TITLE
feat: add pagination to endpoint

### LINKED ISSUE #288: Introduce Pagination
### 🙁 Actual behavior

Our Response take quite long, due to the huge payloads, when displaying all records

### 🙂 Expected behaviour

We should add pagination to make the transferred payload more lightweight

### 🔢 Steps to Reproduce the Problem

_No response_

### PR DESCRIPTION
**Pull request by Cahllagerfeld**
closes #288

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/api/pull/289"></a>

### CODE REVIEW

**Review by Cahllagerfeld — COMMENTED** [review]
We prob need to add that to other endpoints as well

File: test/step-definitions/responses.ts [hunk 1]
```diff
@@ -108,4 +108,38 @@ export class responses {
...
+            expect(isValidID).to.be.true;
+            break;
+          default:
+            const regex = getRegex(data[key]);
+            expect(
+              JSON.parse(this.context.response.text)[property][index][key],
+            ).to.match(regex);
+            break;
+        }
+      } else {
```
Comment (eddiejaoude) [inline test/step-definitions/responses.ts:138]: I find it easier to read if the  `if` ends with a `return` and then the `else` is not required

Comment (Cahllagerfeld) [reply test/step-definitions/responses.ts:138]: You got me there :D this test was generated by Copilot xD I'll refactor it 👍

**Review by eddiejaoude — APPROVED** [review]
LGTM 👍

**Pull request merged**